DuckDuckGo/PasswordGenerationPromptView.swift

```diff
@@ -18,67 +18,69 @@
+            AutofillViews.CloseButtonHeader(action: viewModel.cancelButtonPressed)
+                .offset(x: AutofillViews.isIPhonePortrait(verticalSizeClass, horizontalSizeClass) ? Const.Size.closeButtonOffsetPortrait
+                                                                                                           : Const.Size.closeButtonOffset)
+                .zIndex(1)
+
+                VStack {
+                    AutofillViews.Headline(title: UserText.autofillPasswordGenerationPromptTitle)
+                        .padding(.top, Const.Size.topPadding)
+                    if #available(iOS 16.0, *) {
+                        passwordView
+                            .padding([.top, .bottom], passwordVerticalPadding)
+                    } else {
+                        Spacer()
+                        passwordView
+                        Spacer()
                     }
-                    .buttonStyle(.plain) // Prevent taps from being forwarded to the container view
+                    AutofillViews.Description(text: UserText.autofillPasswordGenerationPromptSubtitle)
+                    contentViewSpacer
+                    ctaView
+                        .padding(.bottom, AutofillViews.isIPad(verticalSizeClass, horizontalSizeClass) ? Const.Size.bottomPaddingIPad
+                                                                                                                : Const.Size.bottomPadding)
                 }
-                Spacer()
-                Text(UserText.autofillPasswordGenerationPromptSubtitle)
-                        .font(Const.Fonts.subtitle)
-                        .secondaryTextStyle()
-                        .multilineTextAlignment(.center)
-                        .padding(.horizontal, isSmallFrame ? Const.Size.paddingSmallDevice : Const.Size.paddingDefault)
-                        .fixedSize(horizontal: false, vertical: true)
-                Spacer()
-                ctaView
-                Spacer()
-                    .frame(height: 24)
-            }
+                .background(GeometryReader { proxy -> Color in
+                    DispatchQueue.main.async { viewModel.contentHeight = proxy.size.height }
+                    return Color.clear
+                })
+                .useScrollView(shouldUseScrollView(), minHeight: frame.height)
+
         }
+        .padding(.horizontal, AutofillViews.isIPhonePortrait(verticalSizeClass, horizontalSizeClass) ? Const.Size.closeButtonOffsetPortrait
+                                                                                                              : Const.Size.closeButtonOffset)
+
+    }
+
+    private func shouldUseScrollView() -> Bool {
+        var useScrollView: Bool = false
+
+        if #available(iOS 16.0, *) {
+            useScrollView = viewModel.contentHeight > UIScreen.main.bounds.size.height - (UIApplication.shared.windows.first?.safeAreaInsets.top ?? 0)
```

Comment:
`.windows` was deprecated on iOS15.  You may want to use  `UIWindowScene.windows` here if as we likely need to support multiple windows on iPad sometime